### ISSUE TITLE
Crash while sendind error with HttpSender

### ISSUE THREAD

**Issue by nicofossa** [ACRA/acra#626]
Hi,
I just migrated my app to the new 5.0.1 version, coming from 4.6.1.
I followed the Wiki page about migration but I can't still receive the logs.

This is the configuration:

```
@acracore(sharedPreferencesName = "it.nicofossa.", reportFormat = StringFormat.KEY_VALUE_LIST)
@acratoast(resText = R.string.crash_toast_text)
@acrahttpsender(uri = "http:///*******.php",
httpMethod = HttpSender.Method.POST)
```

And here it is the logcat output of the ACRA process:

```
01-28 10:06:23.842 16903-16924/it.nicofossa.:acra E/ACRA: Failed to send crash report for /data/user/0/it.nicofossa./app_ACRA-approved/2018-01-28T10:06:20.648-05:00-IS_SILENT.stacktrace
org.acra.sender.ReportSenderException: Policy marked this task as incomplete. ACRA will try to send this report again.
at org.acra.sender.ReportDistributor.sendCrashReport(ReportDistributor.java:121)
at org.acra.sender.ReportDistributor.distribute(ReportDistributor.java:76)
at org.acra.sender.SenderService.onHandleIntent(SenderService.java:92)
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:76)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:164)
at android.os.HandlerThread.run(HandlerThread.java:65)
Caused by: org.acra.sender.ReportSenderException: Error while sending KEY_VALUE_LIST report via Http POST
at org.acra.sender.HttpSender.send(HttpSender.java:172)
at org.acra.sender.ReportDistributor.sendCrashReport(ReportDistributor.java:109)
at org.acra.sender.ReportDistributor.distribute(ReportDistributor.java:76) 
at org.acra.sender.SenderService.onHandleIntent(SenderService.java:92) 
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:76) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loop(Looper.java:164) 
at android.os.HandlerThread.run(HandlerThread.java:65) 
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
at java.net.URLEncoder.encode(URLEncoder.java:204)
at org.acra.data.StringFormat$2.append(StringFormat.java:80)
at org.acra.data.StringFormat$2.toFormattedString(StringFormat.java:66)
at org.acra.sender.HttpSender.convertToString(HttpSender.java:229)
at org.acra.sender.HttpSender.send(HttpSender.java:162)
at org.acra.sender.ReportDistributor.sendCrashReport(ReportDistributor.java:109) 
at org.acra.sender.ReportDistributor.distribute(ReportDistributor.java:76) 
at org.acra.sender.SenderService.onHandleIntent(SenderService.java:92) 
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:76) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loop(Looper.java:164) 
at android.os.HandlerThread.run(HandlerThread.java:65) 
```

Any idea?

**Comment by HelmutSeitz** [comment]
I was seeking for hours, but now I saw, I get the same crash.

**Comment by F43nd1r** [comment]
Fixed with 5.0.2, released today

**Comment by nicofossa** [comment]
Great, tested, it works now!
Thank you!
